src/Akka.Persistence.Sql.Linq2Db/Query/Dao/BaseByteReadArrayJournalDAO.cs

```diff
@@ -126,32 +126,22 @@ private static int MaxTake(long max)
             var maxTake = MaxTake(max);
             return AsyncSource<JournalRow>.FromEnumerable(new{separator,tag,offset,maxOffset,maxTake,_connectionFactory},
                     async(input)=>
-                {
+                    {
+                    var tag = $"{separator}{input.tag}{separator}";
```

Comment:
Tags were matched as "{separator}{tag}{separator}" so that we can be sure that the whole tag were matched

Reply:
Good catch